src/main/scala/com/bazaarvoice/sswf/service/WorkflowManagement.scala

```diff
@@ -0,0 +1,264 @@
+   * @param runId The particular run of the workflow id to list events for
+   * @return
+   */
+  def describeExecution(workflowId: String, runId: String): StepsHistory[SSWFInput, StepEnum] = {
+    val request: GetWorkflowExecutionHistoryRequest = new GetWorkflowExecutionHistoryRequest()
+       .withDomain(domain)
+       .withExecution(new model.WorkflowExecution().withWorkflowId(workflowId).withRunId(runId))
+
+    val iterateFn = (prev: History) => {
+      if (prev == null | prev.getNextPageToken == null) null
+      else swf.getWorkflowExecutionHistory(request.withNextPageToken(prev.getNextPageToken))
+    }
+
+    val historyEvents =
+      Stream
+         .iterate(swf.getWorkflowExecutionHistory(request))(iterateFn)
+         .takeWhile(_ != null)
+         .flatten(_.getEvents)
+         .toList
+
+    HistoryFactory.from(historyEvents, inputParser)
+  }
+
+
+  private[this] def registerDomain(): Unit = {
+    // Opt for try/catch rather than list(), since the missing exception will probably be thrown once for each domain. List, on the other hand, would be expensive every time.
+    try {
+      val domainDetail: DomainDetail = swf.describeDomain(new DescribeDomainRequest().withName(domain))
+      assert(domainDetail.getDomainInfo.getStatus == "REGISTERED", s"domain[$domain] is not REGISTERED [${domainDetail.getDomainInfo.getStatus}]. Please delete it and then re-run.")
+    } catch {
+      case e: UnknownResourceException =>
+        try {
+          swf.registerDomain(new RegisterDomainRequest()
+             .withName(domain)
+             .withDescription(s"domain[$domain] created by SSWF at [${new Date()}]")
+             .withWorkflowExecutionRetentionPeriodInDays(workflowExecutionRetentionPeriodInDays.toString))
+        } catch {
+          case d: DomainAlreadyExistsException =>
+            // race condition. ignore...
+            ()
+        }
+    }
+  }
+
+  private[this] def registerWorkflowType(): Unit = {
+    try {
+      val workflowType: WorkflowTypeDetail = swf.describeWorkflowType(new DescribeWorkflowTypeRequest().withDomain(domain).withWorkflowType(new WorkflowType().withName(workflow).withVersion
+         (workflowVersion)))
+      assert(workflowType.getTypeInfo.getStatus == "REGISTERED", s"workflow[$workflow/$workflowVersion] is not REGISTERED [${workflowType.getTypeInfo.getStatus}]. Please delete it and then re-run.")
+    } catch {
+      case e: UnknownResourceException =>
+        try {
+          swf.registerWorkflowType(new RegisterWorkflowTypeRequest()
+             .withDomain(domain)
+             .withName(workflow)
+             .withVersion(workflowVersion)
+             .withDescription(s"workflow[$workflow/$workflowVersion] registered by SSWF at [${new Date()}}]")
+             .withDefaultExecutionStartToCloseTimeout(workflowExecutionTimeoutSeconds.toString)
+             .withDefaultTaskStartToCloseTimeout(60.toString) // timeout for decision tasks
```

Comment:
This is how long it takes us to make a decision after we poll for one. I don't see that taking longer than a minute, but do you think I should make it configurable? I'm trying to limit your exposure to crap you don't need to worry about.